ok I'am sorry

Gr3g027 opened the pull request:
YES

Commit:
Divided the env dictionary (bot,mega)
Added skin_uploader.py, and warining color

Commit:
Added mega.py in requirements.txt

Commit:
Removed pychache

Commit:
Fixed skins path

Sclafus commented on skin_uploader.py:
```diff
@@ -0,0 +1,42 @@
+   skin_folder_path = os.path.join(osu_path, os.path.join('Skins', skin_name))
+
+mega_file = m.find(f"{skin_name}.osk")
+mega_folder = m.find('Osu! Skins')
```
this "osu skins" folder should be customizable

Sclafus commented on skin_uploader.py:
```diff
@@ -0,0 +1,42 @@
+mega_folder = m.find('Osu! Skins')
+
+if mega_file:
+    print(f"{colors.YELLOW}The skin is already on mega!")
```
This should return the URL for the skin

Sclafus reviewed (changes requested):
ok cool

Sclafus commented on skin_uploader.py:
```diff
@@ -0,0 +1,42 @@
+    for root, dirs, files in os.walk(f"{skin_folder_path}"):
+        for file in files:
+            zip_skin.write(os.path.join(root, file))
+    zip_skin.close()
```
you should use this syntax:
```python
with ZipFile(f"{skin_name}.osk", "w") as zf:
	# stuff
```
instead of opening and closing the zip file manually

Gr3g027 replied:
thx boss

Sclafus reviewed (changes requested):
forgot about this

Commit:
- skin_uploader.py
Defined upload_skin function (idk if it's right, but it works)
Customizable Mega folder
Refactored zipping section
Returned the skin link
- twitch_chatbot.py
Added import from skin_uploader.py
Added skin url (needs a shortener)
- __init__.py
Added mega skin folder to env

Sclafus commented on twitch_chatbot.py:
```diff
@@ -37,34 +38,36 @@ def get_skin() -> dict:
     if api_data:
 
         skin_name = api_data['settings']['folders']['skin']
+        skin_url = upload_skin()
```
ew naming, should be `get_skin()` or something

Sclafus commented on twitch_chatbot.py:
```diff
@@ -87,7 +90,9 @@ async def skin(ctx):
         skin_stripped = " ".join(skin['skin'].split())
-        await ctx.send(f"/me Right now {env['CHANNEL']} is using {skin_stripped}")
+        
+        # TODO find a way to shorten the skin url
```
maybe [bit.ly](https://dev.bitly.com) api? should also add an option in .env to enable the shortener

something like `BITLY_SHORTENER = True`

Sclafus commented on skin_uploader.py:
```diff
@@ -0,0 +1,49 @@
+        os.remove(f"{skin_name}.osk")
+
+        return public_link
+    else: 
```
this `else` statement is unnecessary, the `if` condition has a `return`

Sclafus reviewed (changes requested):
nice work, just some polishing comments

Commit:
Changed function name from 'upload_skin' to 'get_skin_url'
Removed that else in skin_uploader.py

Gr3g027 committed:
Merge branch 'Sclafus:senpai' into senpai

Commit:
Merge branch 'senpai' of https://github.com/sianogregorio/twitch-chatbot-for-gosumemory into senpai

Commit:
Added cache to speed up the skin command
